lib/down/net_http.rb

```diff
@@ -231,10 +237,15 @@ def net_http_request(uri, options, follows_remaining:, &block)
           raise ResponseError.new("Invalid Redirect URI: #{response["Location"]}", response: response)
         end
 
+        # do not leak credentials on redirect
+        uri.user = nil unless auth_on_redirect
+        uri.password = nil unless auth_on_redirect
```

Comment:
do i need this? or asked another way, should credentials stay when it's a relative redirect?

Reply:
going to remove if it's not a relative redirect